faust/sensors/monitor.py

```diff
@@ -135,6 +139,15 @@ class Monitor(ServiceProxy, Sensor, KeywordReduce):
     #: Arbitrary counts added by apps
     metric_counts: Counter[str] = cast(Counter[str], None)
 
+    #: Last committed offsets by TopicPartition
+    tp_committed_offsets: TPOffsetMapping = cast(TPOffsetMapping, None)
+
+    #: Last read offsets by TopicPartition
+    tp_read_offsets: TPOffsetMapping = cast(TPOffsetMapping, None)
+
+    #: Log end offsets by TopicPartition
+    tp_end_offsets: TPOffsetMapping = cast(TPOffsetMapping, None)
```

Comment:
Oh, haha. I was initially confused by the diff here.  The diff makes it look like these are new members of the TableState class.

I think periodically updating this tp_end_offset high-water can be problematic since the number will be outdated
by the time we use it in the monitor.  It will be an approximate number.

Reply:
Yeah but I think its the best we can do as the log end offset is unrelated to this particular application. The application shouldn't be keeping track of it.